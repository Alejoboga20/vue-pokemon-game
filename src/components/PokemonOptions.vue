<template>
	<ul>
		<li v-for="pokemon in pokemons" @click="$emit('selection', pokemon.id)">
			{{ pokemon.name }}
		</li>
	</ul>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import type { PropType } from 'vue';

interface Pokemon {
	id: number;
	name: string;
}

export default defineComponent({
	name: 'PokemonOptions',
	props: {
		pokemons: {
			type: Array as PropType<Pokemon[]>,
			required: true,
		},
	},
});
</script>

<style lang="css" scoped>
ul {
	list-style-type: none;
	margin-top: 64px;
	padding-inline-start: 0;
}

li {
	background-color: white;
	border-radius: 5px;
	border: 1px solid rgba(0, 0, 0, 0.2);
	cursor: pointer;
	margin-bottom: 10px;
	text-align: center;
	width: 100%;
	padding: 4px;
}

li:hover {
	background-color: rgba(0, 0, 0, 0.05);
}
</style>
