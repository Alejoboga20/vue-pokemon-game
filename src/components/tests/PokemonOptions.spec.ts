import { VueWrapper, shallowMount } from '@vue/test-utils';
import PokemonOptions from '@/components/PokemonOptions.vue';
import { pokemonsArray } from '@/mocks/pokemons.mock';

describe('PokemonOptions', () => {
	let wrapper: VueWrapper<any>;

	beforeEach(() => {
		wrapper = shallowMount(PokemonOptions, {
			props: { pokemons: [...pokemonsArray] },
		});
	});

	test('should match the Snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot();
	});
});
