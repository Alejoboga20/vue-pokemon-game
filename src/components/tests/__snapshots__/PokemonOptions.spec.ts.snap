// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PokemonOptions > should match the Snapshot 1`] = `
"<ul data-v-39189cb9=\\"\\">
  <li data-v-39189cb9=\\"\\">bulbasaur</li>
  <li data-v-39189cb9=\\"\\">ivysaur</li>
  <li data-v-39189cb9=\\"\\">venusaur</li>
  <li data-v-39189cb9=\\"\\">charmander</li>
</ul>"
`;
